lib/sgf/gtp_writer.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+class SGF::GtpWriter < SGF::Writer
+  attr_writer :upside_down
+
+  def initialize
+    @upside_down = false
+  end
+
+  private
+
+  def gtp_move(node)
+    pps = node.properties
+    if pps['SZ']
+      @boardsize = pps['SZ'].to_i
+      out = []
+      out << "komi #{pps['KM']}" if pps['KM']
+      out << "boardsize #{pps['SZ']}\nclear_board"
+      pps['AB'].each {|pos| out << to_play("B", pos) } if pps['AB']
+      out.join("\n")
+    elsif pps.size > 1 || (pps.keys != ["B"] && pps.keys != ["W"])
```

Comment:
I'm worried that this means the code will skip any nodes that may have comments on it. Am I misunderstanding the code?

Reply:
you're right, nodes with comments would be ignored.

fixed